### PR TITLE
[9.0][point_of_sale] various fixes

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Description of the issue/feature this PR addresses:

Various errors appearing during the migration of module 'point_of_sale', at post-migration stage.

Current behavior before PR:
Errors stop the migration.

Desired behavior after PR is merged:
Migration of point_of_sale occurs without disruption

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@pedrobaeza @sergio-teruel

**Review by sergio-teruel — APPROVED** [review]

File: addons/point_of_sale/migrations/9.0.1.0.1/post-migration.py [hunk 1]
```diff
@@ -12,6 +12,14 @@
         ('session_id', None)
     ],
 }
+barcode_rule_types_map = {
+    'openupgrade_legacy_9_0_barcode_weight': 'weight',
```
Comment (pedrobaeza) [inline addons/point_of_sale/migrations/9.0.1.0.1/post-migration.py]: Use `openupgrade.get_legacy_name('field')` instead

**Comment by pedrobaeza** [comment]
Please squash commits and don't merge until you are sure this is the definitive solution

**Comment by NL66278** [comment]
:+1:  Tested the changes and solved also my problems with migrating the OpenUpgrade test database from 8.0 to 9.0. Please take the remarks of @pedrobaeza in account, and then it will be ready to merge IMHO.

**Comment by StefanRijnhart** [comment]
After changing the hardcoded field names to a call to get_legacy_name, we can squash the commits when merging.

**Comment by NL66278** [comment]
:+1: Seems ready to merge now...

**Pull request merged**